File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let hashMap = new Map()
        let result = []

        for(let num of nums){
            hashMap.set(num, (hashMap.get(num) || 0) + 1)
        }

        let values = []
        for(const [key, val] of hashMap.entries()){
            values.push([key, val])
        }

        values.sort((a, b) => b[1] - a[1])

        for(let key of values.slice(0, k)){
            result.push(key[0])
        }

        return result
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hashSet = set(nums)
        current = 0
        length = 0
        longest = 0
        
        for num in hashSet:
            if num - 1 in hashSet:
                continue
            
            if num - 1 not in hashSet:
                current = num
                length = 1

                while current + 1 in hashSet:
                    current = current + 1
                    length += 1
            
            if length > longest:
                longest = length

        return longest
        